Exception when shutting down nodes

When the `shutdown()` method of `NodeMainExecutor` is called, an exception is thrown _(only)_ if any of the running nodes is connected to a topic that has subscribers: 
```
May 24, 2017 6:01:09 PM org.ros.internal.node.client.Registrar onPublisherRemoved
INFO: Unregistering publisher: Publisher<PublisherDefinition<PublisherIdentifier<NodeIdentifier</joint_publisher, http://10.34.0.70:42274/>, TopicIdentifier</rosout>>, Topic<TopicIdentifier</rosout>, TopicDescription<rosgraph_msgs/Log, acffd30cd6b6de30f120938c17c593fb>>>>
May 24, 2017 6:01:09 PM org.jboss.netty.channel.DefaultChannelPipeline
WARNING: An exception was thrown by a user handler while handling an exception event ([id: 0x05fce1f1, /10.34.0.70:54329 :> /10.34.0.70:37488] EXCEPTION: java.lang.IllegalStateException: No publisher for topic: /talker)
org.ros.exception.RosRuntimeException: java.lang.IllegalStateException: No publisher for topic: /talker
	at org.ros.internal.transport.ConnectionTrackingHandler.exceptionCaught(ConnectionTrackingHandler.java:81)
	at org.jboss.netty.channel.Channels.fireExceptionCaught(Channels.java:533)
	at org.jboss.netty.channel.AbstractChannelSink.exceptionCaught(AbstractChannelSink.java:49)
	at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:296)
	at org.jboss.netty.handler.codec.frame.FrameDecoder.unfoldAndFireMessageReceived(FrameDecoder.java:458)
	at org.jboss.netty.handler.codec.frame.FrameDecoder.callDecode(FrameDecoder.java:439)
	at org.jboss.netty.handler.codec.frame.FrameDecoder.messageReceived(FrameDecoder.java:311)
	at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:268)
	at org.jboss.netty.channel.Channels.fireMessageReceived(Channels.java:255)
	at org.jboss.netty.channel.socket.nio.NioWorker.read(NioWorker.java:91)
	at org.jboss.netty.channel.socket.nio.AbstractNioWorker.processSelectedKeys(AbstractNioWorker.java:373)
	at org.jboss.netty.channel.socket.nio.AbstractNioWorker.run(AbstractNioWorker.java:247)
	at org.jboss.netty.channel.socket.nio.NioWorker.run(NioWorker.java:35)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalStateException: No publisher for topic: /ur_plugin/joint_states
	at com.google.common.base.Preconditions.checkState(Preconditions.java:149)
	at org.ros.internal.transport.tcp.TcpServerHandshakeHandler.handleSubscriberHandshake(TcpServerHandshakeHandler.java:94)
	at org.ros.internal.transport.tcp.TcpServerHandshakeHandler.messageReceived(TcpServerHandshakeHandler.java:67)
	... 13 more
```

It looks like some TCP connection handler is still active and shutting down the node leaves something in an illegal state.
The line throwing the exception is this one: 
https://github.com/rosjava/rosjava_core/blob/kinetic/rosjava/src/main/java/org/ros/internal/transport/tcp/TcpServerHandshakeHandler.java#L94

--- a/rosjava/src/main/java/org/ros/internal/node/DefaultNode.java
+++ b/rosjava/src/main/java/org/ros/internal/node/DefaultNode.java
@@ -39,8 +39,6 @@ import org.ros.internal.node.service.ServiceDeclaration;
 import org.ros.internal.node.service.ServiceFactory;
 import org.ros.internal.node.service.ServiceIdentifier;
 import org.ros.internal.node.service.ServiceManager;
-import org.ros.internal.node.topic.DefaultPublisher;
-import org.ros.internal.node.topic.DefaultSubscriber;
 import org.ros.internal.node.topic.PublisherFactory;
 import org.ros.internal.node.topic.SubscriberFactory;
 import org.ros.internal.node.topic.TopicDeclaration;
@@ -405,6 +403,7 @@ public class DefaultNode implements ConnectedNode {
     // NOTE(damonkohler): We don't want to raise potentially spurious
     // exceptions during shutdown that would interrupt the process. This is
     // simply best effort cleanup.
+    slaveServer.shutdown();
     topicParticipantManager.shutdown();
     for (ServiceServer<?, ?> serviceServer : serviceManager.getServers()) {
       try {

--- a/rosjava_helpers/src/main/java/org/ros/helpers/ParameterLoaderNode.java
+++ b/rosjava_helpers/src/main/java/org/ros/helpers/ParameterLoaderNode.java
@@ -40,7 +40,7 @@ import java.util.Map;
 public class ParameterLoaderNode extends AbstractNodeMain {
 
     public static final String NODE_NAME = "parameter_loader";
-    private final List<LoadedResource> params = new ArrayList<>();
+    private final List<LoadedResource> params = new ArrayList<LoadedResource>();
     private Log log;
 
     /**
@@ -57,6 +57,7 @@ public class ParameterLoaderNode extends AbstractNodeMain {
         this.params.add(new LoadedResource((new Yaml()).load(ymlInputStream), namespace));
     }
 
+    @SuppressWarnings("unchecked")
     private void addParams(ParameterTree parameterTree, String namespace, Map<String, Object> params) {
         for (Map.Entry<String, Object> e : params.entrySet()) {
             String fullKeyName = namespace + "/" + e.getKey();

